blob/service.go

```diff
@@ -66,15 +80,21 @@ func DefaultSubmitOptions() *SubmitOptions {
 	}
 }
 
-// Submit sends PFB transaction and reports the height in which it was included.
+// Submit sends PFB transaction and reports the height at which it was included.
 // Allows sending multiple Blobs atomically synchronously.
 // Uses default wallet registered on the Node.
 // Handles gas estimation and fee calculation.
-func (s *Service) Submit(ctx context.Context, blobs []*Blob, options *SubmitOptions) (uint64, error) {
+func (s *Service) Submit(ctx context.Context, blobs []*Blob, gasPrice GasPrice) (uint64, error) {
 	log.Debugw("submitting blobs", "amount", len(blobs))
 
-	if options == nil {
-		options = DefaultSubmitOptions()
+	options := DefaultSubmitOptions()
+	if gasPrice >= 0 {
+		blobSizes := make([]uint32, len(blobs))
+		for i, blob := range blobs {
+			blobSizes[i] = uint32(len(blob.Data))
+		}
+		options.GasLimit = blobtypes.EstimateGas(blobSizes, appconsts.DefaultGasPerBlobByte, auth.DefaultTxSizeCostPerByte)
+		options.Fee = types.NewInt(int64(math.Ceil(float64(gasPrice) * float64(options.GasLimit)))).Int64()
```

Comment:
Is there a reason this logic needs to live in blob module rather than deeper down in state? Like state module can have a concept of GasPrice

What I'm concerned about mainly is that `Submit` won't be used by many due to wanting to create and sign the tx remotely rather than via the node's account.

Reply:
Submit should have the easiest ux and SubmitPayForBlob should have full optionality, which is why this "optionality" is calculated here from the easy ux option of gasPrice

Reply:
The point is, we should have GasPrice at StateModule level too so users can use it more comfortable (like even for basic transfers). We can make this into a separate issue.

Reply:
Error: error: No such logger